application/account-management/Tests/Api/Users/UserEndpointsTests.cs

```diff
@@ -74,6 +74,88 @@ public async Task GetUser_WhenInvalidUserId_ShouldReturnBadRequest()
         await EnsureErrorStatusCode(response, HttpStatusCode.BadRequest, $"""Failed to bind parameter "UserId Id" from "{invalidUserId}".""");
     }
     
+    [Fact]
+    public async Task GetUsers_WhenSearchingBasedOnUserEmail_ShouldReturnUser()
+    {
+        // Arrange
+        var searchEmail = string.Concat(DatabaseSeeder.User1ForSearching.Email.Take(4));
+        
+        // Act
+        var response = await TestHttpClient.GetAsync($"/api/users?search={searchEmail}");
+        
+        // Assert
+        EnsureSuccessGetRequest(response);
+        var userResponse = await DeserializeResponse<GetUsersResponseDto>(response);
+        userResponse.Should().NotBeNull();
+        userResponse!.TotalCount.Should().Be(1);
+        userResponse.Users.First().Email.Should().Be(DatabaseSeeder.User1ForSearching.Email);
+    }
+    
+    [Fact]
+    public async Task GetUsers_WhenSearchingBasedOnUserFirstName_ShouldReturnUser()
+    {
+        // Arrange
+        var searchEmail = string.Concat(DatabaseSeeder.User1ForSearching.FirstName!.Take(4));
```

Comment:
I guess this should be `var firstName`. Any reason to take a substring and not just do:

`var firstName = DatabaseSeeder.User1ForSearching.FirstName;`

In which case it could be inlined.

Reply:
You could make a name a complex name like "William Henry Gates" and then search for "Will". As a read of the test that would be clearer, as I inherently understand the intent.

Reply:
This is totally fine. More like a real use case.